src/include/duckdb/common/progress_bar.hpp

```diff
@@ -0,0 +1,60 @@
+//===----------------------------------------------------------------------===//
+//                         DuckDB
+//
+// duckdb/common/progress_bar.hpp
+//
+//
+//===----------------------------------------------------------------------===//
+
+#pragma once
+
+#include <iostream>
+#include <thread>
```

Comment:
Both the <thread> and <future> includes, and all that depends on them, should be hidden behind an #ifndef DUCKDB_NO_THREADS